pkg/operator/ceph/webhook.go

```diff
@@ -63,6 +68,151 @@ func isSecretPresent(ctx context.Context, context *clusterd.Context) (bool, erro
+	ctx := context.TODO()
+	var lastPod corev1.Pod
+
+	for i := 0; i < 30; i++ {
+		pods, err := clusterdContext.Clientset.CoreV1().Pods("cert-manager").List(ctx, options)
+
+		logger.Info("checking if cert-manager pods are running")
+		lastStatus := ""
+		running := 0
+
+		if err == nil && len(pods.Items) > 0 {
+			for _, pod := range pods.Items {
+				if pod.Status.Phase == "Running" {
+					running++
+				}
+				lastPod = pod
+				lastStatus = string(pod.Status.Phase)
+			}
+			if running == len(pods.Items) {
+				logger.Infof("All %d pod(s) with label %s are running", len(pods.Items), label)
+				break
+			}
+		}
+		logger.Infof("waiting for pod(s) with label %s in namespace %s to be running. status=%s, running=%d/%d, err=%+v",
+			label, namespace, lastStatus, running, len(pods.Items), err)
+		time.Sleep(1 * time.Second)
+	}
+	if len(lastPod.Name) == 0 {
+		logger.Infof("no pod was found with label %s", label)
+	}
+	// for i := 0; i < 30; i++ {
+	// 	p, _ := clusterdContext.Clientset.AdmissionregistrationV1().ValidatingWebhookConfigurations().List(ctx, options)
+	// 	logger.Info("INSDIDE validating webhook loop")
+	// 	for _, ValidatingWebhookConfiguration := range p.Items {
+	// 		validatingWebhook := ValidatingWebhookConfiguration.Webhooks
+	// 		for _, i := range validatingWebhook {
+	// 			if ca := i.ClientConfig.CABundle; ca != nil {
+	// 				logger.Info("BUNDLE IS EMPTY")
+	// 			} else {
+	// 				logger.Info("BUNDLE IS NOT EMPTY")
+	// 				break
+	// 			}
+	// 		}
+	// 	}
+	// 	time.Sleep(1 * time.Second)
+	// }
+
+	logger.Info("defining issuer")
+
+	createIssuer := &cmapi.Issuer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "selfsigned-issuer",
+			Namespace: "rook-ceph",
+		},
+		Spec: cmapi.IssuerSpec{
+			IssuerConfig: cmapi.IssuerConfig{
+				SelfSigned: &cmapi.SelfSignedIssuer{},
+			},
+		},
+	}
+	logger.Info("done defining issuer")
+
+	logger.Info("creating issuer")
+	issuer, err := clusterdContext.CmClient.CertmanagerV1().Issuers("rook-ceph").Create(ctx, createIssuer, metav1.CreateOptions{})
+	logger.Info(err)
+	logger.Info("created issuer")
+	logger.Info("done defining certificate")
+
+	createCert := &cmapi.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-admission-controller-cert",
+			Namespace: "rook-ceph",
+		},
+		Spec: cmapi.CertificateSpec{
+			DNSNames: []string{
+				"rook-ceph-admission-controller",
+				"rook-ceph-admission-controller.rook-ceph.svc",
+				"rook-ceph-admission-controller.rook-ceph.cluster.local",
+			},
+			IssuerRef: cmmeta.ObjectReference{
+				Kind: "Issuer",
+				Name: issuer.Name,
+			},
+			SecretName: "rook-ceph-admission-controller",
+		},
+	}
+	logger.Info("done defining certificate")
+	logger.Info("creating certificate")
+	_, _ = clusterdContext.CmClient.CertmanagerV1().Certificates(namespace).Create(ctx, createCert, metav1.CreateOptions{})
+	logger.Info("crated certificate")
+	createValidatingWebhook()
+
+}
+func createValidatingWebhook() {
```

Comment:
hmm creating a webhook is a permission I don't think the operator has, and we probably don't want to add it since it's a higher security risk.